src/modules/job-exec/exec_config.c

```diff
@@ -283,14 +289,19 @@ int config_init (flux_t *h, int argc, char **argv)
         /* Finally, override values on cmdline */
         for (int i = 0; i < argc; i++) {
             if (strstarts (argv[i], "job-shell="))
-                exec_conf.default_job_shell = argv[i]+10;
+                tmpconf.default_job_shell = argv[i]+10;
             else if (strstarts (argv[i], "imp="))
-                exec_conf.flux_imp_path = argv[i]+4;
+                tmpconf.flux_imp_path = argv[i]+4;
             else if (strstarts (argv[i], "service="))
-                exec_conf.exec_service = argv[i]+8;
+                tmpconf.exec_service = argv[i]+8;
         }
     }
 
+    exec_conf.default_job_shell = tmpconf.default_job_shell;
+    exec_conf.flux_imp_path = tmpconf.flux_imp_path;
+    exec_conf.exec_service = tmpconf.exec_service;
+    exec_conf.exec_service_override = tmpconf.exec_service_override;
+    exec_conf.sdexec_properties = tmpconf.sdexec_properties;
```

Comment:
tmpconf is on the local stack, do you think it should be made global?  I guess I could `memcpy()` it instead??

Reply:
No I was just proposing you use structure assignment instead of manually assigning each member, e.g.
```diff
diff --git a/src/modules/job-exec/exec_config.c b/src/modules/job-exec/exec_config.c
index 4e206b6b5..286138491 100644
--- a/src/modules/job-exec/exec_config.c
+++ b/src/modules/job-exec/exec_config.c
@@ -298,11 +298,7 @@ int config_setup (flux_t *h,
         }
     }
 
-    exec_conf.default_job_shell = tmpconf.default_job_shell;
-    exec_conf.flux_imp_path = tmpconf.flux_imp_path;
-    exec_conf.exec_service = tmpconf.exec_service;
-    exec_conf.exec_service_override = tmpconf.exec_service_override;
-    exec_conf.sdexec_properties = tmpconf.sdexec_properties;
+    exec_conf = tmpconf;
     return 0;
 }
 
```

Reply:
my memory must be hazy or I'm thinking of ancient compilers, i could have swore in my head this wasn't safe.

Reply:
AFAICT this is just normal structure assignment and is equivalent to what was there, but won't have to change each time things are added to the struct.